ompi/tools/mpirun/main.c

```diff
@@ -78,6 +78,32 @@ int main(int argc, char *argv[])
         if (NULL != opal_install_dirs.bindir) {
             pfx = strdup(opal_install_dirs.bindir);
         }
+#else
+    } else {
+        char *tmp, *t2;
+        /* for external PRRTE, use the full path to prterun and set the lib */
+        if (NULL != PRTE_PATH) {
+            tmp = opal_basename(PRTE_PATH);
+            if (NULL != tmp && 0 != strcmp(tmp, "prterun")) {
+                /* this is a pure path */
+                asprintf(&pfx, "%s/bin", PRTE_PATH);
+                free(tmp);
+                tmp = opal_dirname(PRTE_PATH);
+            } else {
+                pfx = opal_dirname(PRTE_PATH);
+                /* we need to move up one more level to get to the lib directory */
+                t2 = opal_dirname(pfx);
+                asprintf(&tmp, "%s/lib", t2);
```

Comment:
Shouldn't we use the specified PRTE libdir here, rather than assuming that it is $prefix/lib?  Also, I could be reading the code wrong, but it looks like we're potentially adding /usr/lib to LD_LIBRARY_PATH, which doesn't seem like the right thing to do here.